modules/data-entry/src/main/frontend/src/questionnaire/FileResourceQuestion.jsx

```diff
@@ -0,0 +1,320 @@
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+import { Grid, IconButton, LinearProgress, Link, TextField, Typography, withStyles } from "@material-ui/core";
+import Delete from "@material-ui/icons/Delete";
+
+import PropTypes from "prop-types";
+
+import Answer from "./Answer";
+import DragAndDrop from "../dragAndDrop";
+import { useFormReaderContext } from "./FormContext";
+import { useFormUpdateWriterContext } from "./FormUpdateContext";
+import Question from "./Question";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+import AnswerComponentManager from "./AnswerComponentManager";
+
+// Component that renders a file upload question.
+// Filepaths are placed in a series of <input type="hidden"> tags for
+// submission.
+//
+// Optional arguments:
+//  namePattern (optional): a string specifying if the filename can be processed to extract the subject ID from it
+//
+// Sample usage:
+// (TODO)
+function FileResourceQuestion(props) {
+  const { classes, existingAnswer, ...rest } = props;
+  const { maxAnswers, minAnswers, namePattern } = { ...props.questionDefinition, ...props }
+  let initialValues =
+    // Check whether or not we have an initial value
+    (!existingAnswer || existingAnswer[1].value === undefined) ? [] :
+    // The value can either be a single value or an array of values; force it into a dictionary
+    Array.of(existingAnswer[1].value).flat()
+    // Finally, split it into a series of [value, label]s
+    .map((element) => [/.+\/(.+?)$/.exec(element)?.[1], element]);
+  let initialDict = {};
+  initialValues.forEach((value) => {
+    initialDict[value[0]] = value[1];
+  });
+  let [ uploadedFiles, setUploadedFiles ] = useState(initialDict);
+  let [ error, setError ] = useState();
+  let [ uploadInProgress, setUploadInProgress ] = useState(false);
+  let [ answerPath, setAnswerPath ] = useState(existingAnswer);
+
+  // Default value of knownAnswers is the name of every field we can find in namePattern
+  let initialParsedAnswers = {};
+  if (namePattern) {
+    var varNamesRegex = /@{(.+?)}/g;
+    var varNames = [...namePattern.matchAll(varNamesRegex)].map((match) => match[1]);
+
+    var clearedNamesRegex = namePattern.replaceAll(varNamesRegex, "(.+)");
+    var nameRegex = new RegExp(clearedNamesRegex, "g");
+
+    // Match each of the values into our default initialParsedAnswers
+    initialValues.forEach((filename) => {
+      let results = [...filename[0].matchAll(nameRegex)].map((match) => match[1]);
```

Comment:
This seems wrong to me. Say, for example, name pattern is `@{patient}_@{tumor}\.csv` and filename is `p01_liver.csv` then:
- `varNames` is `["patient", "tumor"]` (correct)
- `nameRegex` is `(.*)_(.*)\.csv` (fragile, but correct)
- `filename.matchAll(nameRegex)` produces one match, not two, and `match[1]` will be the first captured group, `p01`, thus `results` is `["p01"]` (wrong).

```suggestion
      let results = filename[0].match(nameRegex).slice(1);
```